### DISCUSSION TITLE
Sellect/deselect all checboxes

### DISCUSSION THREAD

**Discussion by BYEDUCK** [petereon/beaupy#66] [Ideas]
Hello,
is it possible for "select all" feature to be implemented for checkboxes ?
For example: left arrow -> select all, right arrow -> deselect all

**Comment by petereon** [comment]
Hey @BYEDUCK,

This is a very good suggestion, thank you, what's more implementation-wisely this should be trivial. However the example you offer (using right and left arrows) would conflict with pagination functionality included in version `3.5`.

Let me ponder on it, `beaupy` supports control sequences (`CTRL+A` and such) so I might be able to include functionality that could allow selecting all options and marking them at the same time which feels more natural to me.

**Reply by A-nglea** [reply]
Select all or select none would be an extremely useful addition.

**Reply by petereon** [reply]
Hi @A-nglea, thanks for reminding me about this. I put features on hold temporarily as I was going through a major rewrite of the codebase. But now that it's done, this has a green light. I should be able to find time to take a look at this over the next week or two. If you are interested to take a look yourself I am more than happy to provide guidance and accept a pull-request.

**Comment by petereon** [comment]
@A-nglea @BYEDUCK - this is now implemented, took more time than I would have hoped for

**Reply by xxJava** [reply]
Thank you for the implementation. However I am noticing that when I click A it only deselects all but it refuses to select all. I am running Windows 11 23H2.

**Reply by A-nglea** [reply]
Yes @petereon working sweetly, thank you.
